### PR TITLE
Edit chapter 05

### PR DESCRIPTION
**Pull request by edilmedeiros**
Minor text editing.

### CODE REVIEW

File: 05_vectors_and_result_enum.md [hunk 1]
```diff
@@ -9,14 +9,20 @@ fn read_version(transaction_hex: &str) -> u32 {
 ```
 
 There are two different ways we can approach this: 
-1. Read the first 4 pairs of characters and do some base math to determine the u32 integer.
-2. Convert the transaction into a collection bytes first and then just work with the methods available for that type.
+1. read the first 4 pairs of characters and do some base math to determine the u32 integer;
```
Comment (sb1752) [inline 05_vectors_and_result_enum.md]: not familiar with this. Does this need a semicolon at the end?

Comment (edilmedeiros) [reply 05_vectors_and_result_enum.md]: Formally, numbered lists are still single phrases and the colon is not even needed.

The equivalent of 

There are two different ways we can approach this (1) read the first 4 pairs (...), or (2) convert the transaction (...).

Semicolons substitute commas to differentiate those that separate the list elements. It's usual to use them in long items like these even when there are no commas.

See [what APA says](https://apastyle.apa.org/style-grammar-guidelines/lists/lettered) (they are the reference of correct English in the USA for official and academic documents)

Comment (edilmedeiros) [reply 05_vectors_and_result_enum.md]: Oh, wait. 

Just rechecked their manual and I'm conflating lettered and numbered lists. So many "rules"... 

I'll revert this.

File: 05_vectors_and_result_enum.md [hunk 2]
```diff
@@ -57,18 +73,37 @@ fn read_version(transaction_hex: &str) -> u32 {
...
+Remember, we want to work with a Vector of bytes so a `vec<u8>` is the data type we're looking for.
+However, if we look at the return type of the `decode` function we see that the data structure we want is wrapped *inside* of a `Result` type.
 
-The `Result` type is a common `enum` that you will see in Rust code. Enums are a way to describe a mutually exclusive set of options for a particular variable. If you think about it, it's possible that the `hex::decode` function fails to return a proper collection of bytes. For example, what if one of the characters is not a hex character? So we get two possibilities from a `Result`, an `Ok` response or an `Err` response.
+The `Result` type is a common `enum` that you will see in Rust code.
+Enums are a way to describe a mutually exclusive set of values for a particular variable.
+If you think about it, the `hex::decode` function can fail to return a proper collection of bytes.
+For example, what if one of the characters is not a hex character?
+So we get two possibilities from a `Result`, an `Ok` response or an `Err` response.
+The former represents a successful computation; the later represents that an error occurred.
```
Comment (sb1752) [inline 05_vectors_and_result_enum.md]: *latter

Maybe would change this to: "The former represents a successful computation whereas the latter indicates an error occurred"

Comment (edilmedeiros) [reply 05_vectors_and_result_enum.md]: Surely better.

File: 05_vectors_and_result_enum.md [hunk 3]
```diff
@@ -87,22 +88,21 @@ Let's examine what `hex::decode` returns.
...
 
-
 The `Result` type is a common `enum` that you will see in Rust code.
-Enums are a way to describe a mutually exclusive set of options for a particular variable.
-If you think about it, it's possible that the `hex::decode` function fails to return a proper collection of bytes.
+Enums are a way to describe a mutually exclusive set of values for a particular variable.
+If you think about it, the `hex::decode` function can fail to return a proper collection of bytes.
 For example, what if one of the characters is not a hex character?
 So we get two possibilities from a `Result`, an `Ok` response or an `Err` response.
+The former represents a successful computation whereas the latter represents that an error occurred.
```
Comment (sb1752) [inline 05_vectors_and_result_enum.md]: might read easier if you replace the "represents that" with "indicates"
"...whereas the latter indicates an error occurred"

Comment (edilmedeiros) [reply 05_vectors_and_result_enum.md]: Agree

**Comment by sb1752** [comment]
Sweet! Thanks so much 👍

**Pull request merged**